src/components/OrientationAwareIcon.tsx

```diff
@@ -17,32 +18,41 @@ export const OrientationAwareIcon = (props: Props) => {
   const { orientation, dimensions } = useContext(OrientationContext);
   const {
     Icon,
+    iconName,
     iconStyle,
     landscapeIconStyle,
     landscapeStyle,
     size = normalize(24),
     style
   } = props;
 
+  const SelectedIcon = Icon || IconComponent.NamedIcon;
+  const isNamedIcon = SelectedIcon === IconComponent.NamedIcon;
+
   const needLandscapeStyle =
     orientation === 'landscape' || dimensions.width > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH;
 
-  // we need adjustments only on iOS, so otherwise just return the item with the usual props
-  // TODO: is this still needed with Expo 48?
-  if (!needLandscapeStyle || !(device.platform === 'ios')) {
-    return <Icon {...props} />;
-  }
+  const getIconProps = () => {
+    const baseProps = {
+      ...props,
+      ...(isNamedIcon && { name: iconName })
+    };
+
+    if (!(device.platform === 'ios') || !needLandscapeStyle) {
+      return baseProps;
+    }
+
+    const adjustedSize = size * LANDSCAPE_ADJUSTMENT_FACTOR;
```

Comment:
this is also just used once, right? could be directly set for `size` without variable as well

Reply:
Yes, that's right. But I still added it as a comment line so that later on we can see what that number means.

Reply:
hä? the `LANDSCAPE_ADJUSTMENT_FACTOR` was great. i meant t oset `size: size * LANDSCAPE_ADJUSTMENT_FACTOR` directly

Reply:
But we use `adjustedSize` in 2 different places. So I thought we should keep it, so I return the code.

Reply:
which second place do you mean? i have not seen it

Reply:
In line 49 and 50 we use `iconStyle` and `size` props.

```
    const adjustedSize = size * LANDSCAPE_ADJUSTMENT_FACTOR;

    return {
      ...baseProps,
      iconStyle: [iconStyle, styles.marginIcon, { width: adjustedSize }, landscapeIconStyle],
      size: adjustedSize,
      style: [style, landscapeStyle]
    };
```

Reply:
ok thanks!